pkg/receive/config.go

```diff
@@ -37,12 +37,18 @@ const (
 	RouterIngestor ReceiverMode = "RouterIngestor"
 )
 
+type Endpoint struct {
+	Address string `json:"address"`
+	AZ      string `json:"az"`
+}
+
 // HashringConfig represents the configuration for a hashring
 // a receive node knows about.
+// TODO: unmarhsal endpoints also from strings without AZ for backwards compatibility
```

Comment:
You can do this with the following function 
```
func (e *Endpoint) UnmarshalJSON(data []byte) error {
	// First try to unmarshal as a string.
	err := json.Unmarshal(data, &e.Address)
	if err == nil {
		return nil
	}

	// If that fails, try to unmarshal as an endpoint object.
	type endpointAlias Endpoint
	var configEndpoint endpointAlias
	err = json.Unmarshal(data, &configEndpoint)
	if err == nil {
		e.Address = configEndpoint.Address
		e.AZ = configEndpoint.AZ
	}

	return err
}
```

Reply:
Oh, yeah thats better; i did it in hashring config which is worse; ill do this instead, one moment

Reply:
done, thank you